[MKC-1351] OPAMP Docs (#1596)
arduino/docs-content: content/hardware/02.hero/boards/uno-r4-wifi/tutorials/opamp/opamp.md

tags:
  - OPAMP
author: 'Maximilian Gerhardt, Hannes Siebeneicher'
hardware:
  - hardware/02.hero/boards/uno-r4-wifi
software:
  - web-editor
  - ide-v1
  - ide-v2
---

In this tutorial, you will learn how to use the built-in operational amplifier (OPAMP) featured on the [Arduino UNO R4 WiFi](https://store.arduino.cc/products/uno-r4-wifi). Operational amplifiers are very versatile and can be used to e.g. mirror an electrical signal or amplify it. 

## Goals

In this article, you will learn:
- about the OPAMP feature onboard the UNO R4 WiFi,
- about the basics of the OPAMP library, 
- how to mirror a voltage signal,
- how to amplify a voltage signal.

## Hardware & Software Needed
  To follow along with this article, you will need the following hardware: 
  
  - [Arduino UNO R4 WiFi](https://store.arduino.cc/uno-r4-wifi)
  - resistors
  - Jumper wires
  
  
## Operational Amplifier (OPAMP)

An OPAMP is a versatile and widely used electronic component that belongs to the class of analog integrated circuits. Its primary function is to amplify voltage signals but they are very versatile and can be used to:

- mirror an input voltage to its output,

> - amplify a small analog voltage to its output pin, output voltage range from 0 to ~4.7 V,

Author: ```suggestion
- amplify a small analog voltage to its output pin, the UNO R4's output voltage range is 0 to ~4.7 V,
```

- compare two input voltages and give a binary "higher" or "lower" output,
- integrate and differentiate signals.

## Voltage Follower

The simplest way to test the OPAMP is to configure it as a voltage follower by connecting A2 to A3. In this setup, the voltage at A3 should mimic the voltage applied to A1. For instance, if you connect A1 to the ground (GND), the OPAMP output at A3 should also be at the ground potential. Similarly, if you connect A1 to 3.3 V, the output at A3 should be approximately 3.3 V.

## Circuit

![Voltage Follower Circuit](./assets/circuitFollowerWiFi.png)

## Voltage Amplifier

> ***Do NOT amplify the voltage provided by a battery as this can pose serious safety risks!***

Author: ```suggestion
***Caution: When amplifying the voltage of a battery using an operational amplifier, you should be aware of potential risks. The amplification process can lead to a high current draw, essentially placing a significant load on the battery, which may result in overheating, damage, or even pose a security risk. To mitigate this risk, you may for example add a resistor in series with the battery.***
```

> A voltage amplifier, as the name suggests, amplifies the voltage. A simple 2x amplifier can be built using e.g. two `10k` resistors. Connect one resistor between "minus" and GND. Then use the second resistor to connect the output and "minus" together. Any signal input at "plus" will now appear with double the amplitude at the output pin. Of course, the input signal and the Arduino board should share the same `GND`.

Author: ```suggestion
A voltage amplifier, as the name suggests, amplifies the voltage. A simple 2x amplifier can be built using e.g. two `10k` resistors. Connect one resistor between "minus" and GND. Then use the second resistor to connect the output and "minus" together. a circuit diagram [can be seen below](#circuit). Any signal input at "plus" will now appear with double the amplitude at the output pin. Of course, the input signal and the Arduino board should share the same `GND`. 
```

> A voltage amplifier, as the name suggests, amplifies the voltage. A simple 2x amplifier can be built using e.g. two `10k` resistors. Connect one resistor between "minus" and GND. Then use the second resistor to connect the output and "minus" together. a circuit diagram [can be seen below](#circuit). Any signal input at "plus" will now appear with double the amplitude at the output pin. Of course, the input signal and the Arduino board should share the same `GND`. 

Reviewer: ```suggestion
A voltage amplifier, as the name suggests, amplifies the voltage. A simple 2x amplifier can be built using e.g. two `10k` resistors. Connect one resistor between "minus" and GND. Then use the second resistor to connect the output and "minus" together. A circuit diagram [can be seen below](#circuit). Any signal input at "plus" will now appear with double the amplitude at the output pin. Of course, the input signal and the Arduino board should share the same `GND`. 
```

---
title: 'Arduino UNO R4 WiFi OPAMP'
description: 'Learn how to use the built-in Operational Amplifier in the UNO R4 WiFi'
tags:
  - OPAMP
author: 'Maximilian Gerhardt, Hannes Siebeneicher'
hardware:
  - hardware/02.hero/boards/uno-r4-wifi
software:
  - web-editor
  - ide-v1
  - ide-v2
---

In this tutorial, you will learn how to use the built-in operational amplifier (OPAMP) featured on the [Arduino UNO R4 WiFi](https://store.arduino.cc/products/uno-r4-wifi). Operational amplifiers are very versatile and can be used to e.g. mirror an electrical signal or amplify it. 

## Goals

In this article, you will learn:
- about the OPAMP feature onboard the UNO R4 WiFi,
- about the basics of the OPAMP library, 
- how to mirror a voltage signal,
- how to amplify a voltage signal.

## Hardware & Software Needed
  To follow along with this article, you will need the following hardware: 
  
  - [Arduino UNO R4 WiFi](https://store.arduino.cc/uno-r4-wifi)

>   - resistors 

Reviewer: What value?

Author: Depends on the amplification. I think the section explaining how to calculate the resistor value should be helpful.

  - Jumper wires
  
  
## Operational Amplifier (OPAMP)

An OPAMP is a versatile and widely used electronic component that belongs to the class of analog integrated circuits. Its primary function is to amplify voltage signals but they are very versatile and can be used to:

- mirror an input voltage to its output,
- amplify a small analog voltage to its output pin, output voltage range from 0 to ~4.7 V,
- compare two input voltages and give a binary "higher" or "lower" output,
- integrate and differentiate signals.

## Voltage Follower

> The most simple way of testing the OPAMP is to create a voltage follower, by connecting `A2` to `A3`. Then, any voltage applied to `A1` should be mirrored on `A3`. For example, if you connect `A1` to `GND`, the OPAMP output should be `GND`. Connect `A1` to `3.3 V`, the output should be `3.3 V`.

Reviewer: ```suggestion
The simplest way to test the OPAMP is to configure it as a voltage follower by connecting A2 to A3. In this setup, the voltage at A3 should mimic the voltage applied to A1. For instance, if you connect A1 to the ground (GND), the OPAMP output at A3 should also be at the ground potential. Similarly, if you connect A1 to 3.3 V, the output at A3 should be approximately 3.3 V.
```
